File: fibonacci/fibonacci.scala
object Fibonacci {
    def fibonacci(n: Int): Int = {
        if (n == 0 || n == 1) 1
        else fibonacci(n - 1) + fibonacci(n - 2)
    }

    def main(args: Array[String]): Unit = {
        val n: Int = 42
        val result: Int = fibonacci(n)
        println(result)
    }
}

File: fibonacci/fibonacci.rs
fn fibonacci(n: i32) -> i32 {
    if n == 0 || n == 1 {
        1
    } else {
        fibonacci(n - 1) + fibonacci(n - 2)
    }
}

fn main() {
    let n = 42;
    let result = fibonacci(n);
    println!("{}", result);
}
